src/data/int/basic.lean

```diff
@@ -1312,14 +1317,26 @@ end int
 
 open int
 
-theorem add_monoid_hom.eq_int_cast {A} [add_group A] [has_one A] (f : ℤ →+ A) (h1 : f 1 = 1)
-  (n : ℤ) : f n = n :=
-begin
-  have : ∀ n : ℕ, f n = n, from λ n, (f.comp of_nat_hom.to_add_monoid_hom).eq_nat_cast h1 n,
-  cases n,
-  { exact this n },
-  rw [cast_neg_succ_of_nat, neg_succ_of_nat_eq, f.map_neg, f.map_add, h1, this]
-end
+namespace add_monoid_hom
+
+variables {A : Type*}
+
+/-- Two additive monoid homomorphisms `f`, `g` from `ℤ` to an additive monoid are equal
+if `f 1 = g 1`. -/
+@[ext] theorem ext_int [add_monoid A] {f g : ℤ →+ A} (h1 : f 1 = g 1) : f = g :=
```

Comment:
Should this be done for arbitrary cyclic groups first?

Reply:
I would prefer to say something like "the subgroup closure `{1}` in `int` is `top`, so we're done" but `data/int/basic` is a dependency of `submonoid`.

Reply:
Ok, then we just prove it twice. It's not like it's that much work. I'll review this again. If I have no more comments, we can merge this.